test/DynamoCoreWpfTests/WorkspaceSaving.cs

```diff
@@ -940,6 +940,41 @@ public void CusotmNodeSaveAsUpdateItsName()
                 Assert.AreEqual(node.ID, newId);
             }
         }
+
+        [Test]
+        [Category("UnitTests")]
+        public void CustomNodeWorkspaceSavedToSamePlaceNotCausingDuplicatedLibraryItems()
+        {
+            // open file
+            // save to the temporary folder
+            // save to another temporary folder
+            // save to the old temporary folder
+            // there should be only two library items instead of three
+
+            var dynamoModel = ViewModel.Model;
+            var nodeName = Guid.NewGuid().ToString();
+            var catName = "Custom Nodes";
+
+            var def = dynamoModel.CustomNodeManager.CreateCustomNode(nodeName, catName, "");
+
+            var tempPath1 = Path.Combine(TempFolder, nodeName + ".dyf");
+            var tempPath2 = Path.Combine(Path.Combine(TempFolder, nodeName), nodeName + ".dyf");
```

Comment:
Minor thing but I'd recommend doing `Path.Combine(TempFolder, nodeName, nodeName + ".dyf")` since it's more straightforward.